File: VB/HideNavigationItemsExample.Module/Controllers/CustomShowNavigationItemController.vb
Imports DevExpress.ExpressApp.SystemModule
Imports DevExpress.ExpressApp
Imports HideNavigationItemsExample.Module.Security

Namespace HideNavigationItemsExample.Module.Controllers
	Public Class CustomShowNavigationItemController
		Inherits ShowNavigationItemController

		Protected Overrides Sub SynchItemWithSecurity(ByVal item As DevExpress.ExpressApp.Actions.ChoiceActionItem)
			If Not SecuritySystem.IsGranted(New NavigationItemPermissionRequest(item.Id)) Then
				item.Active(SecurityVisibleKey) = False
			Else
				MyBase.SynchItemWithSecurity(item)
			End If
		End Sub

		Protected Overrides Function SyncItemsWithRequestSecurity(ByVal items As DevExpress.ExpressApp.Actions.ChoiceActionItemCollection) As Boolean
			MyBase.SyncItemsWithSecurity(items)
			Return True
		End Function
	End Class
End Namespace

File: CS/HideNavigationItemsExample.Module/Controllers/CustomShowNavigationItemController.cs
using DevExpress.ExpressApp.SystemModule;
using DevExpress.ExpressApp;
using HideNavigationItemsExample.Module.Security;

namespace HideNavigationItemsExample.Module.Controllers {
    public class CustomShowNavigationItemController : ShowNavigationItemController {
        protected override void SynchItemWithSecurity(DevExpress.ExpressApp.Actions.ChoiceActionItem item) {
            if (!SecuritySystem.IsGranted(new NavigationItemPermissionRequest(item.Id))) {
                item.Active[SecurityVisibleKey] = false;
            } else {
                base.SynchItemWithSecurity(item);
            }
        }
    }
}
